### ISSUE TITLE
no shadows on android compatibility version.

### ISSUE THREAD

**Issue by justlimeguy** [godotengine/godot#83671]
### Godot version

4.1.2 beta 1

### System information

Android 12, OpenGl

### Issue description

so, I tried the engine's shadow feature ( LightOccluder2D with point and directional light 2D) but as soon a i press shadows it just hides the light and does not show any thing. 

- the thing on windows 

https://github.com/godotengine/godot-proposals/assets/142280135/65c17068-9f97-4e0c-a555-1cc07f868cd4

- the thing on android 

https://github.com/godotengine/godot-proposals/assets/142280135/c320371d-11ba-4af0-ba4b-15a2ff8221fa

### Steps to reproduce

just added a light occluder 2d , a color rect and a point light / directional light 2d

### Minimal reproduction project

[Light sample.zip](https://github.com/godotengine/godot/files/13054027/Light.sample.zip)

**Comment by clayjohn** [comment]
Shadows werent added for the compatibility renderer until 4.2 Dev6 (https://github.com/godotengine/godot/pull/77496). They won't be available in a stable version until 4.2. 

Closing as a duplicate of https://github.com/godotengine/godot/issues/66596

**Comment by AThousandShips** [comment]
This is for 2D shadows, does the same apply?

**Comment by justlimeguy** [comment]
@clayjohn but it works on the computer compatibility renderer..
how does it separate itself from android?

**Comment by clayjohn** [comment]
> This is for 2D shadows, does the same apply?

Oops! No, my comment does not apply. I responded too quickly. Since this is for 2D shadows, it is a duplicate of https://github.com/godotengine/godot/issues/75594

> @clayjohn but it works on the computer compatibility renderer..
> how does it separate itself from android?

My apologies, I responded too quickly. 3D shadows were recently added, but 2D shadows still have a bug. The original bug report is here: https://github.com/godotengine/godot/issues/75594.
